File: backend/db/models/meal_types.py
# Meal type constants
class MealType:
    BREAKFAST = 'morgenmad'
    LUNCH = 'frokost'
    DINNER = 'aftensmad'
    SNACK = 'snack'
    SNACK1 = 'mellemmaaltid1'
    SNACK2 = 'mellemmaaltid2'
    
    # Core meal types (4) - used in food header dropdown
    CORE_TYPES = [BREAKFAST, LUNCH, DINNER, SNACK]
    
    # Extended meal types (5) - used in add food and diary
    EXTENDED_TYPES = [BREAKFAST, LUNCH, DINNER, SNACK1, SNACK2]
    
    # All meal types
    ALL_TYPES = [BREAKFAST, LUNCH, DINNER, SNACK, SNACK1, SNACK2]
    
    @classmethod
    def is_valid(cls, meal_type):
        return meal_type in cls.ALL_TYPES
    
    @classmethod
    def get_display_name(cls, meal_type):
        display_names = {
            cls.BREAKFAST: 'Morgenmad',
            cls.LUNCH: 'Frokost',
            cls.DINNER: 'Aftensmad',
            cls.SNACK: 'Snack',
            cls.SNACK1: 'Mellemmåltid 1',
            cls.SNACK2: 'Mellemmåltid 2'
        }
        return display_names.get(meal_type, meal_type)
    
    @classmethod
    def get_core_types(cls):
        return cls.CORE_TYPES
    
    @classmethod
    def get_extended_types(cls):
        return cls.EXTENDED_TYPES

File: frontend/src/data/meal_types.js
// Central definition of meal types
// This is the single source of truth for meal types across the application

export const MEAL_TYPES = {
    BREAKFAST: 'morgenmad',
    LUNCH: 'frokost', 
    DINNER: 'aftensmad',
    SNACK: 'snack',
    SNACK1: 'mellemmaaltid1',
    SNACK2: 'mellemmaaltid2'
};

export const MEAL_DISPLAY_NAMES = {
    [MEAL_TYPES.BREAKFAST]: 'Morgenmad',
    [MEAL_TYPES.LUNCH]: 'Frokost',
    [MEAL_TYPES.DINNER]: 'Aftensmad', 
    [MEAL_TYPES.SNACK]: 'Snack',
    [MEAL_TYPES.SNACK1]: 'Mellemmåltid 1',
    [MEAL_TYPES.SNACK2]: 'Mellemmåltid 2'
};

// All meal types as array
export const ALL_MEAL_TYPES = Object.values(MEAL_TYPES);

// Core meal types (for food header dropdown)
export const CORE_MEAL_TYPES = [
    MEAL_TYPES.BREAKFAST,
    MEAL_TYPES.LUNCH,
    MEAL_TYPES.DINNER,
    MEAL_TYPES.SNACK
];

// Extended meal types (for add food and diary)
export const EXTENDED_MEAL_TYPES = [
    MEAL_TYPES.BREAKFAST,
    MEAL_TYPES.LUNCH,
    MEAL_TYPES.DINNER,
    MEAL_TYPES.SNACK1,
    MEAL_TYPES.SNACK2
];

// Helper functions
export function getMealDisplayName(mealType) {
    return MEAL_DISPLAY_NAMES[mealType] || mealType;
}

export function isValidMealType(mealType) {
    return ALL_MEAL_TYPES.includes(mealType);
}

export function getMealOptions(coreOnly = false) {
    const types = coreOnly ? CORE_MEAL_TYPES : EXTENDED_MEAL_TYPES;
    return types.map(type => ({
        value: type,
        label: getMealDisplayName(type)
    }));
}
